darts/models/forecasting_model.py

```diff
@@ -101,6 +105,343 @@ def _build_forecast_series(self,
+
+        pred_times = [start]
+        while pred_times[-1] <= last_pred_time:
+            pred_times.append(pred_times[-1] + training_series.freq() * stride)
+
+        # iterate and predict pointwise
+        values = []
+        times = []
+
+        iterator = _build_tqdm_iterator(pred_times, verbose)
+
+        if not retrain and not self._fit_called:
+            fit_function(training_series.drop_after(start), target_series.drop_after(start), verbose=verbose)
+
+        for pred_time in iterator:
+            train = training_series.drop_after(pred_time)  # build the training series
+            target = target_series.drop_after(pred_time)  # build the target series
+            if (retrain):
+                fit_function(train, target)
+                pred = self.predict(forecast_horizon, **predict_kwargs)
+            else:
+                pred = self.predict(forecast_horizon, input_series=train, **predict_kwargs)
+            values.append(pred.values()[-1])  # store the N-th point
+            times.append(pred.end_time())  # store the N-th timestamp
+
+        forecast = TimeSeries.from_times_and_values(pd.DatetimeIndex(times), np.array(values))
+
+        return forecast
+
+    @classmethod
+    def gridsearch(model_class,
+                   parameters: dict,
+                   training_series: TimeSeries,
+                   target_series: Optional[TimeSeries] = None,
+                   forecast_horizon: Optional[int] = None,
+                   use_full_output_length: Optional[bool] = None,
+                   val_target_series: Optional[TimeSeries] = None,
+                   num_predictions: int = 10,
+                   use_fitted_values: bool = False,
+                   metric: Callable[[TimeSeries, TimeSeries], float] = metrics.mape,
+                   verbose=False):
+        """ A function for finding the best hyperparameters.
+
+        This function has 3 modes of operation: Expanding window mode, split mode and fitted value mode.
+        The three modes of operation evaluate every possible combination of hyperparameter values
+        provided in the `parameters` dictionary by instantiating the `model_class` subclass
+        of ForecastingModel with each combination, and returning the best-performing model with regards
+        to the `metric` function. The `metric` function is expected to return an error value,
+        thus the model resulting in the smallest `metric` output will be chosen.
+        The relationship of the training data and test data depends on the mode of operation.
+
+        Expanding window mode (activated when `forecast_horizon` is passed):
+        For every hyperparameter combination, the model is repeatedly trained and evaluated on different
+        splits of `training_series` and `target_series`. The number of splits is equal to `num_predictions`, and the
+        forecasting horizon used when making a prediction is `forecast_horizon`.
```

Comment:
Perhaps add a sentence "The `num_predictions` point predictions are obtained by predicting the last `num_predictions` points of the training series with the specified `forecast_horizon`"

Or maybe you'll find a better formulation :D